src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -87,17 +88,33 @@ static Utf16Utility()
                     Vector128<short> vector8800 = Vector128.Create(unchecked((short)0x8800));
                     Vector128<ushort> vectorZero = Vector128<ushort>.Zero;
 
+
+                    Vector128<byte> initialMask = Vector128.Create((ushort)0x1001).AsByte();
+                    Vector128<byte> mostSignficantBitMask = Vector128.Create((byte)0x80).AsByte();
+
                     do
                     {
-                        Vector128<ushort> utf16Data = Sse2.LoadVector128((ushort*)pInputBuffer); // unaligned
+                        Vector128<ushort> utf16Data;
+                        if (AdvSimd.Arm64.IsSupported)
+                        {
+                            utf16Data = AdvSimd.LoadVector128((ushort*)pInputBuffer); // unaligned
+                        }
+                        else
+                        {
+                            utf16Data = Sse2.LoadVector128((ushort*)pInputBuffer); // unaligned
+                        }
                         uint mask;
 
                         Vector128<ushort> charIsNonAscii;
-                        if (Sse41.IsSupported)
-                        {
-                            // Sets the 0x0080 bit of each element in 'charIsNonAscii' if the corresponding
-                            // input was 0x0080 <= [value]. (i.e., [value] is non-ASCII.)
 
+                        // Sets the 0x0080 bit of each element in 'charIsNonAscii' if the corresponding
```

Comment:
This comment should be within both the `if (AdvSimd.Arm64.IsSupported)` and the `else if (Sse41.IsSupported)` blocks. It does not apply to the final `else` block, which is why that block has its own comment.